MdeModulePkg/Core/Dxe/Image/Image.c

```diff
@@ -1189,22 +1221,30 @@ CoreLoadImageCommon (
   }
 
   if (EFI_ERROR (Status)) {
-    Image = NULL;
     goto Done;
   }
 
+  if (gBdsStarted) {
+    ImageOrigin = UefiImageOriginUserImage;
+  }
+
   //
   // Get information about the image being loaded
   //
   Status = UefiImageInitializeContextPreHash (
              &ImageContext,
              FHand.Source,
              (UINT32) FHand.SourceSize,
-             ImageIsFromFv
+             UEFI_IMAGE_SOURCE_FV,
+             ImageOrigin
              );
+  if (Status == EFI_NOT_STARTED) {
+    goto Done;
+  }
+
   if (EFI_ERROR (Status)) {
-    ASSERT (FALSE);
-    return Status;
+    CpuDeadLoop ();
+    return Status; ///< Should be unreachable.

```

Comment:
Although `return Status;` should be unreachable, if we have it at all should it possibly be `goto Done;`? (With the aim, I guess, of doing the sanest thing we would do, if we were unable to deadloop.)

Reply:
Yes, you're right.